components/src/ProfileInfo/profileInfo.main.tsx

```diff
@@ -86,58 +94,22 @@ class ProfileInfoMain extends React.Component<ProfileInfoMainProps, ProfileInfoM
     });
   }
 
-  submitPersonalInfoChange(event) {
+
+  async submitPersonalInfoChange(event) {
     event.preventDefault();
     const {
       firstName, lastName,
     } = this.state;
-    login().then(() => {
-      cortexFetch('/', {
-        headers: {
-          'Content-Type': 'application/json',
-          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
-        },
-      }).then(res => res.json())
-        .then((res) => {
-          const profileNameLink = res.links.find(link => link.rel === 'defaultprofile');
-          cortexFetch(`${profileNameLink.uri}?followlocation=true`, {
-            headers: {
-              'Content-Type': 'application/json',
-              Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
-            },
-          }).then(linkRes => linkRes.json())
-            .then((linkRes) => {
-              cortexFetch(linkRes.self.uri, {
-                method: 'put',
-                headers: {
-                  'Content-Type': 'application/json',
-                  Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
-                },
-                body: JSON.stringify({
-                  'given-name': firstName,
-                  'family-name': lastName,
-                }),
-              }).then((response) => {
-                if (response.status === 400) {
-                  this.setState({ failedSubmit: true });
-                } else if (response.status === 201 || response.status === 200 || response.status === 204) {
-                  this.cancel();
-                  const { onChange } = this.props;
-                  onChange();
-                }
-              }).catch((error) => {
-                // eslint-disable-next-line no-console
-                console.error(error.message);
-              });
-            }).catch((error) => {
-            // eslint-disable-next-line no-console
-              console.error(error.message);
-            });
-        }).catch((error) => {
-        // eslint-disable-next-line no-console
-          console.error(error.message);
-        });
+    const { onChange } = this.props;
+    const root = await this.client.root().fetch({ defaultprofile: {} });
+    this.client.profile(root.defaultprofile.uri).update({
+      givenName: firstName,
+      familyName: lastName,
+      profileId: '',
     });
+    onChange();
+    this.cancel();
+    onChange();
```

Comment:
You are calling onChange() twice. Is this intentional or a mistake?

Reply:
There was some issue with updating givenName and familyName. Sometimes after calling onChange() those fields are not updated. That is why I call onChange() twice. In this case, everything works correctly.